mathesar_ui/src/pages/schemas/DeleteSchema.svelte

```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
+  import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+  import {
+    Button,
+    Modal,
+    Icon,
+  } from '@mathesar-components';
+  import type { Schema } from '@mathesar/App.d';
+  import { States } from '@mathesar/utils/api';
+  import { deleteSchema } from '@mathesar/stores/schemas';
+  import { currentDBName } from '@mathesar/stores/databases';
+
+  export let isOpen = false;
+  export let schema: Schema;
+
+  let state: States = States.Idle;
+  let error;
+
+  async function removeSchema() {
+    if (schema) {
+      try {
+        state = States.Loading;
+        error = null;
+        await deleteSchema($currentDBName, schema.id);
+        state = States.Done;
+      } catch (err) {
+        state = States.Error;
+        error = (err as Error).message;
+      }
+    }
+  }
+</script>
+
+<Modal class="schema-modal" bind:isOpen closeOnEsc={state !== States.Loading}>
+  <div class="header">
+    Delete schema '{schema?.name}'
+  </div>
+
+  {#if schema?.tables.size > 0}
```

Comment:
It would be better to check the schema's `has_dependencies` property here, since there could be other dependencies, not just tables. #398 will provide an API to actually list the dependencies.

Reply:
@kgodey The `has_dependencies` property always seems to return `true`, whether or not the schema has tables. Since we currently only showcase tables, I've checked this condition based on the table size.

After #398 is done we'll have to update this modal to display all dependencies. We can change the condition to check using  `has_dependencies` when we get to that.

Reply:
I was thinking that it would be better to show no dependencies or always show that there are dependencies rather than only show tables. This is because if there are no tables in the schema but there are other dependencies, this message will mislead users into thinking there are no dependencies while confirming deletion of the schema. I would at least add a note about there being other potential dependencies so that we don't mislead users if we end up deprioritizing #398 for some reason.

Reply:
Makes sense. Does this look okay? I've removed displaying the tables, and added a common message, based on `has_dependencies`. If you think we can also show tables for now, I'll include it.

Note: This hasn't been committed yet. I wanted to confirm the UX.

<img width="515" alt="Screenshot 2021-08-18 at 3 45 08 PM" src="https://user-images.githubusercontent.com/11850603/129881211-4acd770a-e9a5-4581-9ef5-0ba8134c2dfc.png">

Reply:
We can show the tables for now. I would say something like "Tables are shown below, but there may be other dependencies that are not shown here that will also be deleted."

Reply:
@kgodey I've committed this change in da3396b6d4aab35d5564461fb315d54947213693